inst/template/mode.R.template

```diff
@@ -133,8 +133,27 @@
     },
 
     has_openmp = function() {
-      mode_{{name}}_has_openmp()
-    }
+      mode_{{name}}_capabilities()[["openmp"]]
+    },
+
+    has_gpu_support = function(fake_gpu = FALSE) {
+      if (fake_gpu) {
```

Comment:
what's the purpose of this `fake_gpu` arg?

Reply:
https://mrc-ide.github.io/dust/reference/dust_generator.html#method-dust-has_gpu_support - so if a model has gpu support at the code level, but not at the object level. sircovid (the covid model) by default has this out the box, so we can test the GPU kernels on the CPU. But if you want to test that it *really* has GPU support you pass `FALSE` here and it checks that it has both the code support and was also compiled with `nvcc` in order to run on the GPU. Hope that makes sense, it's all a bit weird and not well described in the gpu vignette yet it seems (https://mrc-ide.github.io/dust/articles/gpu.html)